pandas/tests/frame/test_arithmetic.py

```diff
@@ -127,132 +126,97 @@ def test_df_add_flex_filled_mixed_dtypes(self):
              'B': ser * 2})
         tm.assert_frame_equal(result, expected)
 
-    def test_arith_flex_frame(self):
-        seriesd = tm.getSeriesData()
-        frame = pd.DataFrame(seriesd).copy()
+    @pytest.mark.parametrize('op', ['add', 'sub', 'mul', 'div', 'truediv',
+                                    'pow', 'floordiv', 'mod'])
+    def test_arith_flex_frame(self, op, int_frame, mixed_int_frame,
+                              float_frame, mixed_float_frame):
 
-        mixed_float = pd.DataFrame({'A': frame['A'].copy().astype('float32'),
-                                    'B': frame['B'].copy().astype('float32'),
-                                    'C': frame['C'].copy().astype('float16'),
-                                    'D': frame['D'].copy().astype('float64')})
-
-        intframe = pd.DataFrame({k: v.astype(int)
-                                 for k, v in seriesd.items()})
-        mixed_int = pd.DataFrame({'A': intframe['A'].copy().astype('int32'),
-                                  'B': np.ones(len(intframe), dtype='uint64'),
-                                  'C': intframe['C'].copy().astype('uint8'),
-                                  'D': intframe['D'].copy().astype('int64')})
-
-        # force these all to int64 to avoid platform testing issues
-        intframe = pd.DataFrame({c: s for c, s in intframe.items()},
-                                dtype=np.int64)
-
-        ops = ['add', 'sub', 'mul', 'div', 'truediv', 'pow', 'floordiv', 'mod']
         if not PY3:
             aliases = {}
         else:
             aliases = {'div': 'truediv'}
-
-        for op in ops:
-            try:
-                alias = aliases.get(op, op)
-                f = getattr(operator, alias)
-                result = getattr(frame, op)(2 * frame)
-                exp = f(frame, 2 * frame)
+        alias = aliases.get(op, op)
+
+        f = getattr(operator, alias)
+        result = getattr(float_frame, op)(2 * float_frame)
+        exp = f(float_frame, 2 * float_frame)
+        tm.assert_frame_equal(result, exp)
+
+        # vs mix float
+        result = getattr(mixed_float_frame, op)(2 * mixed_float_frame)
+        exp = f(mixed_float_frame, 2 * mixed_float_frame)
+        tm.assert_frame_equal(result, exp)
+        _check_mixed_float(result, dtype=dict(C=None))
+
+        # vs mix int
```

Comment:
might be more clear to move these to a separate test (from here down)

Reply:
I don't want to get into orthogonal changes here. I haven't changed anything about this except removing the indentation. This goes for the test split requested below as well.